openedx/core/djangoapps/user_api/api.py

```diff
@@ -472,6 +475,88 @@ def _add_year_of_birth_field(self, form_desc, required=True):
             required=required
         )
 
+    def _add_profession_field(self, form_desc, required=True):
+        """Add a profession field to a form description.
+
+        Arguments:
+            form_desc: A form description
+
+        Keyword Arguments:
+            required (bool): Whether this field is required; defaults to True
+
+        """
+        # Translators: This label appears above a dropdown menu on the registration
+        # form used to select the user's profession
+        profession_label = _(u"Profession")
+
+        profession_options = configuration_helpers.get_value('PROFESSION_OPTIONS')
```

Comment:
Maybe change the structure of the options setting to be a dictionary where the keys are the field names and the values are the list of options. For example:
```
'EXTRA_FIELD_OPTIONS': {
    'profession': ['Doctor', 'Registered Nurse'],
    'specialty': ['Internal Medicine', 'Cardiology']
}
```